### PR TITLE
Harp 4924: Let the map view pan past the antimeridian

### PR DESCRIPTION
**Pull request by nzjony**

### CODE REVIEW

**Comment by nzjony** [comment]
This commit allows the user to pan past the given current restriction of -180 <-> 180 degrees longitude.

**Comment by nzjony** [comment]
Please review.

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts [hunk 1]
```diff
@@ -64,28 +64,33 @@ export class MercatorProjection implements Projection {
...
+        geoPointLike: GeoCoordinatesLike,
+        result?: WorldCoordinates,
+        normalize: boolean = true
     ): WorldCoordinates {
-        let normalized: GeoCoordinates;
+        let geoPoint: GeoCoordinates;
 
-        if (geoPoint instanceof GeoCoordinates) {
-            normalized = geoPoint.normalized();
+        if (geoPointLike instanceof GeoCoordinates) {
```
Comment (StefanDachwitz) [inline @here/harp-geoutils/lib/projection/MercatorProjection.ts]: How about:
normalize = !(normalize === false);

File: @here/harp-geoutils/lib/projection/WebMercatorProjection.ts [hunk 2]
```diff
@@ -47,27 +48,25 @@ class WebMercatorProjection extends MercatorProjection {
...
-        );
-        const x = clampedX / EarthConstants.EQUATORIAL_CIRCUMFERENCE - 0.5;
-        const clampedY = MercatorProjection.clamp(
-            worldPoint.y,
-            0,
-            EarthConstants.EQUATORIAL_CIRCUMFERENCE
-        );
+    unprojectPoint(worldPoint: Vector3Like, clamp?: boolean): GeoCoordinates {
+        const finalX =
+            clamp === undefined || clamp === true
```
Comment (StefanDachwitz) [inline @here/harp-geoutils/lib/projection/WebMercatorProjection.ts]: !(clamp !== false)

File: @here/harp-geoutils/lib/projection/Projection.ts [hunk 3]
```diff
@@ -29,8 +29,8 @@ export interface Projection {
     ): Bounds;
 
     /**
-     * Projects a point from geo coordinates (latitude, longitude, altitude) to world coordinates
-     * (x,y,z).
+     * Projects a point from geo coordinates (latitude, longitude, altitude) to
```
Comment (StefanDachwitz) [inline @here/harp-geoutils/lib/projection/Projection.ts]: The "standing order" (see confluence) is to use the line length of 100 characters for comments, and to maintain a block-paragraph-style. This is more readable, and if we omit  this rule, and let everybody choose his favorite way of breaking lines, all hell breaks loose. Literally.

File: @here/harp-mapview/lib/MapView.ts [hunk 4]
```diff
@@ -1489,13 +1490,14 @@ export class MapView extends THREE.EventDispatcher {
     /**
      * Returns the screen position of the given geo coordinates.
      *
-     * @param geoPos The geo coordinates.
-     * @returns The screen position in CSS/client coordinates (no pixel ratio applied) or
-     * `undefined`.
+     * @param geoPos The geo coordinates, value isn't normalized (i.e. clamped
```
Comment (StefanDachwitz) [inline @here/harp-mapview/lib/MapView.ts]: Please follow the comment about line length in all your changes.

File: @here/harp-geoutils/lib/projection/Projection.ts [hunk 5]
```diff
@@ -89,8 +95,10 @@ export interface Projection {
      * ```
      *
      * @param worldBox The bounding box in world coordinates.
+     * @param clamp Whether to clamp the world points, if not clamped, the
```
Comment (StefanDachwitz) [inline @here/harp-geoutils/lib/projection/Projection.ts]: I would like to suggest to indent the lines in the @param section:

@param clamp Whether to clamp the world points, if not clamped, the .... 100 chars are not enough....
-> -> indent by two tabs

This seems to make the code more readable. Just a _suggestion_...

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts (hunk 1 above)
Comment (nzjony) [reply @here/harp-geoutils/lib/projection/MercatorProjection.ts]: looks like Hax, but I guess it is a standard way of converting undefined & false to false?

File: @here/harp-geoutils/lib/projection/Projection.ts (hunk 3 above)
Comment (nzjony) [reply @here/harp-geoutils/lib/projection/Projection.ts]: Thanks for pointing out, my rewrap comments plugin was configured to 80 chars, apologies.

File: @here/harp-geoutils/lib/projection/Projection.ts (hunk 5 above)
Comment (nzjony) [reply @here/harp-geoutils/lib/projection/Projection.ts]: I will remove this parameter, but will take your suggestion into consideration in future.

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts (hunk 1 above)
Comment (StefanDachwitz) [reply @here/harp-geoutils/lib/projection/MercatorProjection.ts]: it is shorter, that is all. There are so many uses of it if parameters can be undefined

File: @here/harp-geoutils/lib/projection/WebMercatorProjection.ts (hunk 2 above)
Comment (StefanDachwitz) [reply @here/harp-geoutils/lib/projection/WebMercatorProjection.ts]: Type, should read: !(clamp === false)

File: @here/harp-geoutils/lib/projection/MercatorProjection.ts (hunk 1 above)
Comment (nzjony) [reply @here/harp-geoutils/lib/projection/MercatorProjection.ts]: all good

**Pull request merged**